PhysicsTools/PatUtils/plugins/L1PrefiringWeightProducer.cc

```diff
@@ -0,0 +1,478 @@
+  double nonPrefiringProba[3] = {1., 1., 1.};        //0: central, 1: up, 2: down
+  double nonPrefiringProbaJet[3] = {1., 1., 1.};     //0: central, 1: up, 2: down
+  double nonPrefiringProbaPhoton[3] = {1., 1., 1.};  //0: central, 1: up, 2: down
+  double nonPrefiringProbaMuon[3] = {1., 1., 1.};    //0: central, 1: up, 2: down
+
+  for (const auto fluct : {fluctuations::central, fluctuations::up, fluctuations::down}) {
+    for (const auto& photon : thePhotons) {
+      double pt_gam = photon.pt();
+      double eta_gam = photon.eta();
+      if (pt_gam < 20.)
+        continue;
+      if (fabs(eta_gam) < 2.)
+        continue;
+      if (fabs(eta_gam) > 3.)
+        continue;
+      double prefiringprob_gam = getPrefiringRate(eta_gam, pt_gam, h_prefmap_photon, fluct);
+      nonPrefiringProba[fluct] *= (1. - prefiringprob_gam);
+      nonPrefiringProbaPhoton[fluct] *= (1. - prefiringprob_gam);
+    }
+
+    //Now applying the prefiring maps to jets in the affected regions.
+    for (const auto& jet : theJets) {
+      double pt_jet = jet.pt();
+      double eta_jet = jet.eta();
+      double phi_jet = jet.phi();
+      if (pt_jet < 20.)
+        continue;
+      if (fabs(eta_jet) < 2.)
+        continue;
+      if (fabs(eta_jet) > 3.)
+        continue;
+      if (jetMaxMuonFraction_ > 0 && jet.muonEnergyFraction() > jetMaxMuonFraction_)
+        continue;
+      //Loop over photons to remove overlap
+      double nonprefiringprobfromoverlappingphotons = 1.;
+      for (const auto& photon : thePhotons) {
+        double pt_gam = photon.pt();
+        double eta_gam = photon.eta();
+        double phi_gam = photon.phi();
+        if (pt_gam < 20.)
+          continue;
+        if (fabs(eta_gam) < 2.)
+          continue;
+        if (fabs(eta_gam) > 3.)
+          continue;
+        double dR = reco::deltaR(eta_jet, phi_jet, eta_gam, phi_gam);
+        if (dR > 0.4)
+          continue;
+        double prefiringprob_gam = getPrefiringRate(eta_gam, pt_gam, h_prefmap_photon, fluct);
+        nonprefiringprobfromoverlappingphotons *= (1. - prefiringprob_gam);
+      }
+      //useEMpt =true if one wants to use maps parametrized vs Jet EM pt instead of pt.
+      if (useEMpt_)
+        pt_jet *= (jet.neutralEmEnergyFraction() + jet.chargedEmEnergyFraction());
+      double nonprefiringprobfromoverlappingjet = 1. - getPrefiringRate(eta_jet, pt_jet, h_prefmap_jet, fluct);
+
+      if (nonprefiringprobfromoverlappingphotons == 1.) {
+        nonPrefiringProba[fluct] *= nonprefiringprobfromoverlappingjet;
+        nonPrefiringProbaJet[fluct] *= nonprefiringprobfromoverlappingjet;
+      }
+      //If overlapping photons have a non prefiring rate larger than the jet, then replace these weights by the jet one
+      else if (nonprefiringprobfromoverlappingphotons > nonprefiringprobfromoverlappingjet) {
+        if (nonprefiringprobfromoverlappingphotons != 0.) {
```

Comment:
Again, for robustness agains numerical fluctuations if could be preferable
```suggestion
        if (nonprefiringprobfromoverlappingphotons <= 0.) {
```

Reply:
Done.